doc-content/drools-docs/src/main/asciidoc/DecisionEngine/stateless-kie-sessions-con.adoc

```diff
@@ -0,0 +1,141 @@
+
+Stateless KIE sessions are commonly used for the following use cases:
+
+* *Validation*, such as validating that a person is eligible for a mortgage
+* *Calculation*, such as computing a mortgage premium
+* *Routing and filtering*, such as sorting incoming emails into folders or sending incoming emails to a destination
+
+For example, consider the following driver's license data model and sample DRL rule:
+
+.Data model for driver's license application
+[source,java]
+----
+public class Applicant {
+  private String name;
+  private int age;
+  private boolean valid;
+  // Getter and setter methods
+}
+----
+
+.Sample DRL rule for driver's license application
+[source]
+----
+package com.company.license
+
+rule "Is of valid age"
+when
+  $a : Applicant(age < 18)
+then
+  $a.setValid(false);
+end
+----
+
+The `Is of valid age` rule disqualifies any applicant younger than 18 years old. When the `Applicant` object is inserted into the {DECISION_ENGINE}, the {DECISION_ENGINE} evaluates the constraints for each rule and searches for a match. The `"objectType"` constraint is always implied, after which any number of explicit field constraints are evaluated. The variable `$a` is a binding variable that references the matched object in the rule consequence.
+
+[NOTE]
+====
+The dollar sign (`$`) is optional and helps to differentiate between variable names and field names.
+====
+
+In this example, the sample rule and all other files in the `~/resources` folder of the {PRODUCT} Maven project are built with the following configuration:
+
+.Create the KIE container
+[source,java]
+----
+KieServices kieServices = KieServices.Factory.get();
+
+KieContainer kContainer = kieServices.getKieClasspathContainer();
+----
+
+This configuration compiles all the rule files found on the classpath and adds the result of this compilation, a `KieModule` object, in the `KieContainer`.
+
+Finally, the `StatelessKieSession` object is instantiated from the `KieContainer` and is executed against specified data:
+
+.Instantiate the stateless KIE session and enter data
+[source,java]
+----
+StatelessKieSession kSession = kContainer.newStatelessKieSession();
+
+Applicant applicant = new Applicant("Mr John Smith", 16);
+
+assertTrue(applicant.isValid());
+
+ksession.execute(applicant);
+
+assertFalse(applicant.isValid());
+----
+
+In a stateless KIE session configuration, the `StatelessKieSession` object wraps the `KieSession` object instead of extending it, so you do not need to call `fireAllRules()` or call `dispose()` after session invocation. With a stateless KIE session, the `execute()` call acts as a combination method that instantiates the `KieSession` object, adds all the user data and executes user commands, calls `fireAllRules()`, and then calls `dispose()`.
```

Comment:
Sure, I understand that. But this paragraph does give some critical info about fireAllRules and dispose in stateless KIE sessions, etc, in regards to the example code just before it, and what this paragraph explains is what I see asked about more than anything else, KIE sessions wise, according to customer tickets.

How about I just remove the bit about the API and leave the rest, would that work for you?

> In a stateless KIE session configuration, you do not need to call `fireAllRules()` or call `dispose()` after session invocation. With a stateless KIE session, the `execute()` call acts as a combination method that instantiates the `KieSession` object, adds all the user data and executes user commands, calls `fireAllRules()`, and then calls `dispose()`.

Reply:
eh, I guess. It would make more sense if this section came *after* the stateful section, but chicken and egg... We could maybe turn it around like this:

> In a stateless KIE session, the execute() call fires all the rules and then disposes of any leftover state, effectively clearing it for any later executions. It acts as a combination method that instantiates the KieSession object, adds all the user data and executes user commands, calls fireAllRules(), and then calls dispose(). It follows that, as opposed to a stateful session (add sect. ref here), you do not need to call fireAllRules() or call dispose() after session invocation.

Reply:
I see. Okay, changed to this:

>In a stateless KIE session configuration, the `execute()` call acts as a combination method that instantiates the `KieSession` object, adds all the user data and executes user commands, calls `fireAllRules()`, and then calls `dispose()`. Therefore, with a stateless KIE session, you do not need to call `fireAllRules()` or call `dispose()` after session invocation as you do with a stateful KIE session.

Reply:
ok